pkg/cli/publish.go

```diff
@@ -76,13 +80,36 @@ func PublishCmd(ctx context.Context, configFile string, outputRefs string, usePr
 		UseProot:           useProot,
 	}
 
+	// SOURCE_DATE_EPOCH takes priority
+	if v, ok := os.LookupEnv("SOURCE_DATE_EPOCH"); ok {
```

Comment:
Might be better to refactor global options into a global context, kinda like in `cosign`.

Reply:
this is cleaner now that it's using functional options, I'll leave other flag-related improvements to another PR.